Fatras/include/ActsFatras/Digitization/ChannelMerger.hpp

```diff
@@ -0,0 +1,73 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2020 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "ActsFatras/Digitization/DigitizationData.hpp"
+#include <Acts/EventData/ParameterSet.hpp>
+#include <Acts/Surfaces/Surface.hpp>
+#include <Acts/Surfaces/SurfaceError.hpp>
+#include <Acts/Utilities/Helpers.hpp>
+#include <Acts/Utilities/ParameterDefinitions.hpp>
+
+#include <array>
+#include <utility>
+
+namespace ActsFatras {
+
+/// Channel combination to a resulting parameter set.
+///
+struct ChannelMerger {
+  /// Generic implementation of a channel merger, currently only additive
+  /// channel merging
+  ///
+  /// @tparam kParameters parameter pack describing the parameters to smear
+  ///
+  /// @param channels The channels from one cluster
+  ///
+  /// @return A cluster containing the parameter set and cluster size
+  template <typename signal_t, Acts::BoundIndices... kParameters>
+  const std::vector<Channel<signal_t, kParameters...>> merge(
+      const std::vector<Channel<signal_t, kParameters...>>& channels) const {
```

Comment:
This method can be implemented as a simple function.

Reply:
you mean instead of a struct ?

Reply:
Yes. At the moment, the `struct` seems to be just used to namespace the function, so they could be implement directly e.g. as `mergeChannels(...)` and `extractChannelKey(...)` instead.

Reply:
Fixed.